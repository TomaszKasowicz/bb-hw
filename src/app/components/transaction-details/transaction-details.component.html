<div class="category" [style.background-color]="transaction?.categoryCode"></div>
  <div class="item date">{{transaction?.transactionDate | date:'MMM. dd' }}</div>
  <div class="item logo">
    <img *ngIf="transaction?.merchantLogo" [src]="transaction?.merchantLogo" alt="">
  </div>
  <div class="item details">
    <div>
      {{transaction?.merchant}}
    </div>
    <div>
      {{transaction?.transactionType}}
    </div>
  </div>
  <div class="item amount">-{{transaction.amount | currency}}</div>    


