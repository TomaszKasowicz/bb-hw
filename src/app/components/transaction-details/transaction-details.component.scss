.container {
  font-size: 1em;
  height: 7em;
  display: flex;
  align-items: center;
  align-content: center;
  background-color: white;
  border-bottom: 1px solid #999990;
}

.category {
  padding: 0;
  height: 100%;
  flex-basis: 1em;
}

.item {
  padding: 0;
  margin: 0em 1em;
}

.details {
  div:first-child {
    font-weight: bold;
  }
  flex-grow: 1;
}

.logo {
  width: 4em;
  height: 4em;
  img {
    max-height: 100%;
    max-width: 100%;
    filter: grayscale(100%);
  }
}
