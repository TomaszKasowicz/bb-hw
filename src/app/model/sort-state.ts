export interface SortState {
  property: string;
  ascending: boolean;
}
