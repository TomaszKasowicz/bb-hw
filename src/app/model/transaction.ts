export interface Transaction {
  amount: number;
  categoryCode: string;
  merchant: string;
  merchantLogo?: string;
  transactionDate: number;
  transactionType: 'Card Payment' | 'Online Transfer' | 'Transaction';
}

const merchants = [
  'The Tea Lounge',
  'Texaco',
  'Amazon Online Store',
  '7-Eleven',
  'H&M Online Store',
  'Jerry Hildreth',
  'Lawrence Pearson',
  'Whole Foods',
  'Southern Electric Company'
];
