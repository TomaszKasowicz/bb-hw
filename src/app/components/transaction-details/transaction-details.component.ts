import { Component, Input, OnInit } from '@angular/core';
import { Transaction } from '../../model/transaction';

@Component({
  selector: 'app-transaction-details',
  templateUrl: './transaction-details.component.html',
  styleUrls: ['./transaction-details.component.scss']
})
export class TransactionDetailsComponent implements OnInit {
  @Input()
  public transaction: Transaction;

  constructor() {}

  ngOnInit(): void {}
}
