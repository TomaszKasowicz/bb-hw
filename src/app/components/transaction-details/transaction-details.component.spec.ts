import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { TransactionDetailsComponent } from './transaction-details.component';
import { Transaction } from '../../model/transaction';

describe('TransactionDetailsComponent', () => {
  let spectator: Spectator<TransactionDetailsComponent>;
  const createComponent = createComponentFactory(TransactionDetailsComponent);
  const transaction: Transaction = {
    amount: 123.45,
    categoryCode: '#123456',
    merchant: 'merchant',
    transactionDate: new Date(2021, 3, 27).getTime(),
    transactionType: 'Transaction',
    merchantLogo: 'Logo'
  };

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should create', () => {
    expect(spectator.component).toBeTruthy();
  });

  it('Should Render Details with logo', () => {
    spectator.setInput({ transaction });
    spectator.detectChanges();
    expect(spectator.fixture).toMatchSnapshot();

    const divElements = spectator.queryAll('.item');
    expect(divElements.length).toEqual(4);

    expect(spectator.query('img')).toBeTruthy();
  });

  it('Should Render Details without logo', () => {
    const transactionWithoutLogo = { ...transaction, merchantLogo: undefined };
    spectator.setInput({ transaction: transactionWithoutLogo });
    spectator.detectChanges();
    expect(spectator.fixture).toMatchSnapshot();

    const divElements = spectator.queryAll('.item');
    expect(divElements.length).toEqual(4);
    expect(spectator.query('img')).toBeFalsy();
  });
});
