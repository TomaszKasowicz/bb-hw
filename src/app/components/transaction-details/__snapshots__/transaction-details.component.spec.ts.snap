// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetailsComponent Should Render Details with logo 1`] = `
<app-transaction-details
  __ngContext__={[Function LRootView]}
  transaction={[Function Object]}
>
  <div
    class="category"
    style="background-color: rgb(18, 52, 86);"
  /><div
    class="item date"
  >
    Apr. 27
  </div><div
    class="item logo"
  >
    <img
      alt=""
      src="Logo"
    />
    
  </div><div
    class="item details"
  >
    <div>
       merchant 
    </div>
    <div>
       Transaction 
    </div>
  </div><div
    class="item amount"
  >
    -$123.45
  </div>
</app-transaction-details>
`;

exports[`TransactionDetailsComponent Should Render Details without logo 1`] = `
<app-transaction-details
  __ngContext__={[Function LRootView]}
  transaction={[Function Object]}
>
  <div
    class="category"
    style="background-color: rgb(18, 52, 86);"
  /><div
    class="item date"
  >
    Apr. 27
  </div><div
    class="item logo"
  >
    
  </div><div
    class="item details"
  >
    <div>
       merchant 
    </div>
    <div>
       Transaction 
    </div>
  </div><div
    class="item amount"
  >
    -$123.45
  </div>
</app-transaction-details>
`;
